lib/wab/impl/configuration.rb

```diff
@@ -42,28 +39,26 @@ def initialize(usage, options)
         @map = merge_map(@map, command_line_map) unless command_line_map.empty?
       end
 
-      # Walks the options map and calls opts.on for each option so that all
-      # are provided when help is called.
-      def add_options(opts, options, path)
+      # Walks the options map and calls +opts.on+ for each option so that all
+      # are provided when +--help+ is called.
+      def add_options(opts, options, path='')
         options.each_pair { |k,v|
           next unless v.is_a?(Hash)
-          key_path = path.empty? ? k.to_s : path + '.' + k.to_s
+          key_path = path.empty? ? k.to_s : "#{path}.#{k}"
```

Comment:
This is really a style thing. The original is as readable and most likely more efficient.

Reply:
I don't think, this is just a style thing.. interpolation is better than concating three strings.. Ruby has to allocate memory for three string objects in the latter case..

Reply:
Try a benchmark. In older Rubies it was the case that concatenation was faster.

Reply:
Tried a benchmark from [`fast-ruby`](https://github.com/JuanitoFatas/fast-ruby/blob/master/code/string/concatenation.rb)
```
Warming up --------------------------------------
            String#+    55.536k i/100ms
       String#concat    55.353k i/100ms
       String#append    54.919k i/100ms
         "foo" "bar"    64.275k i/100ms
  "#{'foo'}#{'bar'}"    64.980k i/100ms
Calculating -------------------------------------
            String#+      1.712M (A± 6.4%) i/s -      8.553M in   5.024036s
       String#concat      1.686M (A± 3.2%) i/s -      8.469M in   5.030116s
       String#append      1.773M (A± 4.5%) i/s -      8.897M in   5.029507s
         "foo" "bar"      2.758M (A± 6.6%) i/s -     13.755M in   5.017857s
  "#{'foo'}#{'bar'}"      2.807M (A± 4.1%) i/s -     14.036M in   5.011824s

Comparison:
  "#{'foo'}#{'bar'}":  2806550.1 i/s
         "foo" "bar":  2758208.8 i/s - same-ish: difference falls within error
       String#append:  1773353.1 i/s - 1.58x  slower
            String#+:  1712461.4 i/s - 1.64x  slower
       String#concat:  1685757.5 i/s - 1.66x  slower
```
on Ruby 2.3

Reply:
Looks like things have changed over the years. I stand corrected.